### PR TITLE
Add Travis configuration to publish doc to GitHub pages

### LINKED ISSUE #89: Failing Documentation
**Describe the bug**
We have doc comments in the code, but we're not appearing on https://docs.rs

I believe we should automatically be added when we publish to Cargo, in this case we are not.

**To Reproduce**
https://docs.rs/releases/search?query=rustscan

**Expected behavior**
We appear on the docs site.

### PR DESCRIPTION
**Pull request by remigourdon**
Closes #89

Hi!

I had a look at the link you shared the other day (which rendered poorly on my end too) and found the original source for the snippet.

It seems possible to get it running without too much hassle. My pull request adds the `after_success` phase to `.travis.yml` and I believe [the proper URL](http://rustscan.github.io/RustScan) for the GitHub pages documentation to the `README.md`.

You will need to [generate a token](https://github.com/settings/tokens) with a `public_repo` scope.

Then store and encrypt the token [in an environment variable](https://docs.travis-ci.com/user/environment-variables/#encrypting-environment-variables) called `GITHUB_TOKEN` with the command below.

```sh
travis encrypt GITHUB_TOKEN=<THE_TOKEN> --add env.global
```

This should have added the proper variable to `.travis.yml` and after committing to this pull request the environment variable should now get populated and the `cargo doc` will get triggered on `master`.

Let me know if there are any issues!

### CODE REVIEW

**Comment by remigourdon** [comment]
Ah yeah of course your Markdown link checker is failing since the documentation page does not exist yet... Not sure how you want to resolve this one.

**Comment by brandonskerritt** [comment]
> Ah yeah of course your Markdown link checker is failing since the documentation page does not exist yet... Not sure how you want to resolve this one.

Ignore the CI, I'll review this in a bit. Thanks so much for making a PR!!! :D

**Comment by brandonskerritt** [comment]
Hey! I can't actually add the token because:
```
Detected repository as remigourdon/RustScan, is this correct? |yes| yes
repository not known to https://api.travis-ci.org/: remigourdon/RustScan**
```

I will merge, but I'll make an issue for you so we can discuss this :)

**Comment by brandonskerritt** [comment]
https://github.com/RustScan/RustScan/issues/142

**Comment by brandonskerritt** [comment]
Do you want me to build in the folder `/docs`? 
image
This is my first time using GH pages :)

**Comment by brandonskerritt** [comment]
Also @all-contributors please add @remigourdon for documentation, website & code

**Comment by brandonskerritt** [comment]
Also @all-contributors please add @remigourdon for documentation, website & code

**Comment by remigourdon** [comment]
Hi @brandonskerritt ! The token issue is odd. I will have a look at this tonight.

For the GitHub pages source, the way its being used in the PR will place the content of  `target/doc/` at the root of the `gh-pages` branch. This is the default branch for GitHub pages and I believe it should get picked up automatically and the pages should become accessible automatically. The settings page is if you need to use another branch / folder as the source.

**Comment by brandonskerritt** [comment]
@CMNatic you know this better than I do please help? :)

**Comment by remigourdon** [comment]
Ok so you should be able to (:crossed_fingers:) specify the proper repository name when adding the token as follow:

```sh
travis encrypt -r "RustScan/RustScan" GITHUB_TOKEN=<THE_TOKEN> --add env.global
```

I think it was trying to auto-detect the slug from my forked repository.

**Comment by brandonskerritt** [comment]
That worked!

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ RustScans **only** job is to reduce the friction between finding open ports and
 
 | Installation Guide | Documentation | Discord |
 | ------------------ | ------------- | ------- |
-| 📖 [Installation Guide](https://github.com/RustScan/RustScan#-full-installation-guide) | 📚 [Documentation](https://github.com/RustScan/RustScan/issues/89) | 🦜 [Discord](https://discord.gg/GFrQsGy)
+| 📖 [Installation Guide](https://github.com/RustScan/RustScan#-full-installation-guide) | 📚 [Documentation](http://rustscan.github.io/RustScan) | 🦜 [Discord](https://discord.gg/GFrQsGy)
```
Comment (SuperSandro2000) [inline README.md:70]:
```suggestion
| 📖 [Installation Guide](https://github.com/RustScan/RustScan#-full-installation-guide) | 📚 [Documentation](https://rustscan.github.io/RustScan/) | 🦜 [Discord](https://discord.gg/GFrQsGy)
```

Comment (remigourdon) [reply README.md]: Thanks @SuperSandro2000. The [GitHub Pages guide's](https://pages.github.com/) example for project pages uses HTTP and not HTTPS, but after double checking the redirect to HTTPS is automatic, so there is no reason indeed to not use it.

**Review by brandonskerritt — APPROVED** [review]

**Comment by brandonskerritt** [comment]
Let's hope this works! :crossed_fingers:

**Comment by remigourdon** [comment]
It doesn't seem to be working :disappointed: The `gh-pages` branch is there, which is odd because it looks like the `after_success` phase hasn't run since the build failed. The Travis interface also doesn't show a `GITHUB_TOKEN` environment variable.

**Comment by brandonskerritt** [comment]
I'll investigate this :)

**Comment by brandonskerritt** [comment]
The token wasn't added, I added it in a PR that got deleted :( I've just pushed it to master :)

**Pull request merged**